fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/expressions/Comparisons.java

```diff
@@ -923,6 +948,13 @@ public Type getType() {
             return type;
         }
 
+        public boolean isCorrelatedTo(@Nonnull final CorrelationIdentifier alias) {
+            if (internal != Bindings.Internal.CORRELATION) {
```

Comment:
Maybe this predicate, already used in `planHash`, should be its own method like `isCorrelation`?

Reply:
Makes sense! Will do that!